flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -130,8 +132,15 @@ public ResultPartition create(
 				bufferCompressor,
 				bufferPoolFactory);
 
+			BiFunction<Integer, PipelinedResultPartition, PipelinedSubpartition> factory;
+			if (type == ResultPartitionType.PIPELINED_APPROXIMATE) {
+				factory = PipelinedApproximateSubpartition::new;
+			} else {
+				factory = PipelinedSubpartition::new;
+			}
+
```

Comment:
nit: I'd prefer this simple ternary if in a loop:
```
for (int i = 0; i < subpartitions.length; i++) {
    subpartitions[i] = type == ResultPartitionType.PIPELINED_APPROXIMATE ?
        new PipelinedApproximateSubpartition(i, pipelinedPartition) :
        new PipelinedSubpartition(i, pipelinedPartition);
}
```

Reply:
LOL, I've changed from this version -> the fancy version based on Arvid's suggestion.